src/viewport.js

```diff
@@ -1518,7 +1519,20 @@ $.Viewport.prototype = {
         var scale = this._contentBoundsNoRotate.width;
         var viewportToImageZoomRatio = (imageWidth / containerWidth) / scale;
         return imageZoom * viewportToImageZoomRatio;
+    },
+
+    /**
+     * Toggle flip state and demands a new drawing on navigator and viewer objects.
+    */
+    toggleFlip: function() {
```

Comment:
This is a great start, but I would also like `setFlip` and `getFlip`, to match what we're doing with rotation. Once those functions exist, `toggleFlip` can call them.

Also please provide doc comments (you can look at the comments above `setRotation` as an example); those comments get turned into the documentation you see at http://openseadragon.github.io/docs/OpenSeadragon.Viewport.html.

Also please fire a `flip` event from `setFlip` (if the value changes). Again, you can look at setRotation for an example.

Reply:
tomorow I will continue this but you were thinking on something like: